### ISSUE TITLE
Ignore pattern not working proprerly

### ISSUE THREAD

**Issue by Njaah-0** [mrmlnc/fast-glob#329]
### Environment

  * OS Version: MacOS Big Sur 11.4
  * Node.js Version: q2.22
  * Fast-glob: 3.2.7

### Actual behavior
Applying pattern that has ! doesn't seem to ignore correctly. When same, reversed pattern is used in ignore option, it works properly
<img width="535" alt="Screenshot 2021-07-28 at 12 15 14" src="https://user-images.githubusercontent.com/14109820/127298490-b26a628b-4193-4fd3-be22-f04bd4377f8c.png">

### Expected behavior
<img width="506" alt="Screenshot 2021-07-28 at 12 15 06" src="https://user-images.githubusercontent.com/14109820/127298527-7c97da63-bbf7-4972-a364-fcd2faaa5dc2.png">

### Steps to reproduce

  1. See the image.

### Code sample

```js
const fg = require('fast-glob');

const entries = fg.sync(['./insertyourtestfolderhere/!(*.d).{ts,tsx}'], {
  dot: true,
  // ignore: ['./insertyourtestfolderhere/(*.d).{ts,tsx}'],
});

console.log(entries);

```

**Comment by leethree** [comment]
i'm having the same issue with pattern `src/**/!(*.test).{ts,tsx}`

**Comment by mrmlnc** [comment]
Looks like a bug in the `micromatch` package.

```js
require('micromatch').makeRe('*!(.stories).js');
// ^(?:(?!\.)(?=.)[^/]*?(?:(?!(?:\.stories))[^/]*?)\.js)$
// match → component.stories.js
```

This issue (https://github.com/micromatch/picomatch/issues/93) is similar to our problem.

**Comment by timtim17** [comment]
I was encountering some quirkiness earlier, and I wasn't sure if it was similar to this issue.

```sh
aus@Austins-MBP src % tree -a
.
├── foo
│   ├── bar.txt
│   └── ignore.txt
├── index.js
└── test
    ├── baz
    │   ├── .not_this_either.txt
    │   └── also_no.txt
    └── no.txt

3 directories, 6 files
```

```js
const fg = require('fast-glob');

const resultsA = fg.sync('**/*.txt', {
    ignore: ['**/test/**/*']
});
console.log(resultsA);

const resultsB = fg.sync('**/[!ignore]*.txt', {
    ignore: ['**/test/**/*']
});
console.log(resultsB);

const resultsC = fg.sync('**/[!ignore]*.txt', {
    ignore: ['**/test/**']
});
console.log(resultsC);
```

```sh
aus@Austins-MBP src % node index.js
[ 'foo/bar.txt', 'foo/ignore.txt' ]
[ 'foo/bar.txt', 'test/baz/.not_this_either.txt' ]
[ 'foo/bar.txt' ]
```

**Results A:** Works as expected. Gets all txt files, ignoring any in the test directory.
**Results B:** Does not work as expected. It does correctly give me all text files in the foo folder, except for the ignored `ignore.txt`, but it also gives me one of (but notably not all) files in the test directory.
**Results C:** Works as expected, after making a slight change to the ignore option.

I only encounter this when an ignore pattern is added to my pattern. If this is working as intended, or not related to this issue, please let me know. Thanks!

**Comment by mrmlnc** [comment]
Awaiting https://github.com/micromatch/picomatch/pull/102.

**Comment by mrmlnc** [comment]
@timtim17, looks like a separate problem. Please fill another issue.

**Comment by mrmlnc** [comment]
Fixed by https://github.com/micromatch/picomatch/pull/102 and released in [2.3.1](https://github.com/micromatch/picomatch/releases/tag/2.3.1). Since this is a transitive dependency on `micromatch`, it is enough to reinstall this package by `npm i fast-glob`.
